### PR TITLE
Fix `tofu validate` with static variables

### PR DESCRIPTION
**Pull request by cam72cam**
Missed during my original PR, added explicit test case.

## Target Release

1.8.0

## Checklist

- [x] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
- [x] I have not used an AI coding assistant to create this PR.
- [x] I have written all code in this PR myself OR I have marked all code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from.

### Go checklist

 

- [x] I have run golangci-lint on my change and receive no errors relevant to my code.
- [x] I have run existing tests to ensure my code doesn't break anything.
- [x] I have added tests for all relevant use cases of my code, and those tests are passing.
- [x] I have only exported functions, variables and structs that should be used from other packages.
- [x] I have added meaningful comments to all exported functions, variables, and structs.

### CODE REVIEW

**Review by ollevche — APPROVED** [review]
Looks good to me, but we need to address command's tech debt at some point

**Comment by cam72cam** [comment]
Agreed.  That's one of the reasons I want to prioritize the evaluation and improvement of testing in the next few release cycles.  It's a complex piece of tech debt.

File: internal/command/graph.go
Comment (RLRabinowitz) [inline internal/command/graph.go:1]: Any reason why we don't need to `GatherVariables` here, but we do have to in pretty much every other command?

Comment (cam72cam) [reply internal/command/graph.go:1]: It depends on the level of refactoring the individual command has seen.  If it uses the arguments package, it's stored in a new Vars structure.  If it's using the legacy Meta calls, it is automatically injected into the meta struct.

File: internal/command/arguments/validate.go [hunk 1]
```diff
@@ -35,10 +37,11 @@ func ParseValidate(args []string) (*Validate, tfdiags.Diagnostics) {
 	var diags tfdiags.Diagnostics
 	validate := &Validate{
 		Path: ".",
+		Vars: &Vars{},
 	}
 
 	var jsonOutput bool
-	cmdFlags := defaultFlagSet("validate")
+	cmdFlags := extendedFlagSet("validate", nil, nil, validate.Vars)
```
[the remark this answers is not in the document]
Comment (cam72cam) [reply internal/command/arguments/validate.go:44]: Was this clarified during our discussion earlier today?  It provides a lot more context in the actual error messages, the tests just look for a small section to make sure it hit the code we expect.

[the remark this answers is not in the document]
Comment (RLRabinowitz) [reply internal/command/arguments/validate.go:44]: Yes this was clarified - The error message are clear indeed.

However, we still need to update the help message for the validate command to include `-var` and `-var-file`

[the remark this answers is not in the document]
Comment (RLRabinowitz) [reply internal/command/arguments/validate.go:44]: We've talked, and you'll handle the help message in a separate PR

[the remark this answers is not in the document]
Comment (cam72cam) [reply internal/command/arguments/validate.go:44]: I'll open up a followup PR for that.

**Pull request merged**